packages/storybook/src/documentation/pages/pagination.stories.mdx

```diff
@@ -90,10 +107,45 @@ To achieve this, just set the pagination property of the `components` react prop
 
 <CustomPaginationDemo />
 
-## Pagination API
+## Server-side Pagination
+
+XGrid supports both client and server side pagination. By default, pagination works on the client side. To switch it to server side,
+set the property `paginationMode` to `Server` as string or you can use our `FeatureMode` enum.
+**Note**: You also need to set the `rowCount` property to override the number of rows in the grid so XGrid can calculate the total number of pages.
+
+Finally, you're going to need to handle the `onPageChanged` event, to load the rows for the corresponding page.
+
+```tsx
+const [rows, setRows] = useState<RowsProp>([]);
+const onPageChanged = useCallback(
+  (params) => {
+    loadServerRows(params).then((newRows) => {
+      setRows(newRows);
+    });
+  },
+  [setRows],
+);
+
+<XGrid
+  rows={rows}
+  columns={columns}
+  options={{
+    pagination: true,
+    pageSize: 5,
+    paginationMode: 'Server',
+    rowCount: 50,
+    onPageChanged: onPageChanged,
+  }}
+/>;
+```
+
+<ServerPaginationDemo />
 
-Now that you understand how to implement the pagination feature of the grid. You might want to manipulate the page or set a listener when the current page change.
-To achieve this, you should use our `apiRef` as below;
+### Pagination API
+
+We exposed a set of methods that will let you achieve all the above features using the `ApiRef` object. More info on our Api page [here](??).
```

Comment:
```suggestion
We exposed a set of methods that will let you achieve all the above features using the `ApiRef` object. More info on our API page [here](??).
```

Reply:
Also please mind https://webaccess.berkeley.edu/ask-pecan/click-here regarding the usage of `[here](??)`.